src/Orleans.Transactions.Azure/Storage/AzureTransactionLogStorage.cs

```diff
@@ -43,29 +43,29 @@ public class AzureTransactionLogStorage : ITransactionLogStorage
         private List<CommitRecord> currentRowTransactions;
         private int currentRowTransactionsIndex;
 
-        public AzureTransactionLogStorage(SerializationManager serializationManager, IOptions<AzureTransactionLogConfiguration> configuration)
+        public AzureTransactionLogStorage(SerializationManager serializationManager, IOptions<AzureTransactionLogOptions> configurationOptions)
         {
             this.serializationManager = serializationManager;
-            this.configuration = configuration.Value;
+            this.options = configurationOptions.Value;
         }
 
         public async Task Initialize()
         {
-            if (string.IsNullOrWhiteSpace(configuration.ConnectionString))
+            if (string.IsNullOrWhiteSpace(this.options.ConnectionString))
             {
-                throw new ArgumentNullException(nameof(configuration.ConnectionString));
+                throw new ArgumentNullException(nameof(this.options.ConnectionString));
```

Comment:
In addition to this, we could add an `IConfigurationValidator` to DI so that we throw on `Build()`. It might help to surface errors even sooner.

Reply:
added validator, but keeping these checks in place.  Component should not assume configuration is valid.